proposals/XXXX-limit-textfield-length-according-to-hmi-capabilities.md

```diff
@@ -0,0 +1,56 @@
+# Limit TextField Length According to HMI Capabilities
+
+* Proposal: [SDL-XXXX](XXXX-limit-textfield-length-according-to-hmi-capabilities.md)
+* Author: [Collin McQueen](https://github.com/iCollin)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+This proposal defines how Core should limit the length of TextFields sent to the HMI according to the HMI Capabilities.
+
+## Motivation
+The Mobile API defines the max length of many text fields much larger than either the SDL HMI or Generic HMI support. The true maximum length of text fields is defined by the HMI Capabilities `displayCapabilities.textFields`. For example, `AddCommand.MenuParams.tertiaryText` is defined with `maxlength="500"` in the Mobile API, but the Generic HMI's capabilities inform Core that it can handle just one twenty character wide row for `tertiaryText`. This means Core may be forwarding TextFields to the HMI that are longer than the HMI can handle.
```

Comment:
```suggestion
The Mobile API defines the max length of many text fields as much larger than either the SDL HMI or Generic HMI support. The true maximum length of text fields is defined by the HMI Capabilities `displayCapabilities.textFields`. For example, `AddCommand.MenuParams.tertiaryText` is defined with `maxlength="500"` in the Mobile API, but the Generic HMI's capabilities inform Core that it can handle just one twenty character wide row for `tertiaryText`. This means Core may be forwarding TextFields to the HMI that are longer than the HMI can handle.
```